rust_src/src/multibyte.rs

```diff
@@ -854,31 +854,30 @@ pub unsafe extern "C" fn str_to_multibyte(
 /// Same as `MULTIBYTE_LENGTH` macro in C.
 #[allow(clippy::if_same_then_else)]
 fn multibyte_length(slice: &[c_uchar], allow_encoded_raw: bool) -> Option<usize> {
-    let len = slice.len();
-    if len < 1 {
-        None
-    } else if slice[0] & 0x80 == 0 {
-        Some(1)
-    } else if len < 2 || slice[1] & 0xC0 != 0x80 {
-        None
-    } else if !allow_encoded_raw && slice[0] & 0xFE == 0xC0 {
-        None
-    } else if slice[0] & 0xE0 == 0xC0 {
-        Some(2)
-    } else if len < 3 || slice[2] & 0xC0 != 0x80 {
-        None
-    } else if slice[0] & 0xF0 == 0xE0 {
-        Some(3)
-    } else if len < 4 || slice[3] & 0xC0 != 0x80 {
-        None
-    } else if slice[0] & 0xF8 == 0xF0 {
-        Some(4)
-    } else if len < 5 || slice[4] & 0xC0 != 0x80 {
-        None
-    } else if slice[0] == 0xF8 && slice[1] & 0xF0 == 0x80 {
-        Some(5)
-    } else {
-        None
+    // The bits a byte "starts" with are the most significant ones in these comments
+    match slice {
+        // true if a starts with 0
+        &[a] if a & 0x80 == 0 => Some(1),
```

Comment:
Don't all of the match patterns have to have wildcards at the end? I think this one should be `&[a, ..]`, for example.

Reply:
That's true, the original code checks for failure conditions rather than success conditions and that has tripped me up a couple of times.